src/components/list/listcontainer.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div class='uk-panel uk-width-1-1'>
+    <ul class='uk-list uk-list-divider'>
+      <list-item
+        v-for='item in items'
+        :key='item.uid'
+        :uid='item.uid'
+        :editor='EditorClass'
+        @delete-item='onDeleteItem'
+        @change-item='onChangeItem'
+      ></list-item>
+    </ul>
+    <div class='uk-panel uk-margin'>
+      <button class="uk-button uk-button-primary uk-float-right"
+      @click="createItem">Add Item</button>
+    </div>
+  </div>
+</template>
+
+<script>
+import Vue from 'vue'
+import ListItem from './listitem.vue'
+import ModalDialog from '../modaldialog.vue'
+
+export default {
+  name: 'list-container',
+  components: {
+    ListItem,
+  },
+  props: {
+    editor: {
+      type: Object,
+      required: true
+    },
+  },
+  data () {
+    return {
+      currentItem: {},
+      ModalDialogClass: Vue.extend(ModalDialog),
+      EditorClass: null,
+      dialogInstance: null,
+      items: {},
+    }
+  },
+  methods: {
+    createItem () {
+      this.currentItem = {}
+      this.dialogInstance = new this.ModalDialogClass({
+        propsData: {
+          component: this.editor,
+          properties: this.currentItem
+        }
+      })
+      this.dialogInstance.$on('cancel', this.onCancelNewItem)
+      this.dialogInstance.$on('save', this.onSaveNewItem)
+      this.dialogInstance.show()
+    },
+    onDeleteItem (key) {
+      let temp = Object.assign({}, this.items)
+      delete temp[key]
+      this.items = temp
+    },
+    onChangeItem (data) {
+      // this.$emit('list-changed', this.serialize())
+    },
+    onCancelNewItem () {
+      this.$set(this, 'dialogInstance', null)
+    },
+    onSaveNewItem (data) {
+      this.$store.dispatch('updateDataSource', data)
```

Comment:
I'd recommend that this be moved to the editor component, so that saving to the store isn't contingent on the editor being displayed from a list.

Reply:
Its not so much that the editor needs to be displayed from a list, but that the editor doesn't handle the saving. Whatever instantiated it can listen to the event.

Again, since the store was added late, the editor isn't aware of the store. That could change, of course. I would mark this as something to refactor.